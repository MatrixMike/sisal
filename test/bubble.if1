T 1 1 0    %na=Boolean
T 2 1 1    %na=Character
T 3 1 2    %na=Double
T 4 1 3    %na=Integer
T 5 1 4    %na=Null
T 6 1 5    %na=Real
T 7 1 6    %na=WildBasic
T 8 10
T 9 0	4    %na=onedim
T 10 8	9	0
T 11 8	4	10
T 12 3	11	10
T 13 3	10	10
T 14 4	9
C$  C Faked IF1CHECK
C$  D Nodes are DFOrdered
C$  F Livermore Frontend   Version1.8
G	12	"Bubble"	%sl=7	%sf=test/bubble.sis
{ Compound   1   4
G	0	%sl=9	%sf=test/bubble.sis
E	0 1	0 3	9	 %na=a	%mk=V	%sl=11
E	0 2	0 4	4	 %na=limit	%mk=V	%sl=11
G	0	%sl=13	%sf=test/bubble.sis
N 1	132	%sl=13	%sf=test/bubble.sis
E	0 4	1 1	4	 %na=limit	%mk=V	%sl=13
L		1 2	4 "1"	%mk=V
N 2	139	%sl=13	%sf=test/bubble.sis
E	1 1	2 1	1	%mk=V
E	2 1	0 1	1	%mk=V
G	0	%sl=15	%sf=test/bubble.sis
N 1	135	%sl=15	%sf=test/bubble.sis
E	0 4	1 1	4	 %na=old-limit	%mk=V	%sl=15
L		1 2	4 "1"	%mk=V
{ Compound   2   4
G	0	%sl=17	%sf=test/bubble.sis
L		0 3	4 "0"	 %na=j	%mk=V	%sl=18
E	0 1	0 4	9	 %na=a1	%mk=V	%sl=19
G	0	%sl=20	%sf=test/bubble.sis
N 1	131	%sl=20	%sf=test/bubble.sis
E	0 3	1 1	4	 %na=j	%mk=V	%sl=20
E	0 2	1 2	4	 %na=limit	%mk=V	%sl=20
E	1 1	0 1	1	%mk=V
G	0	%sl=21	%sf=test/bubble.sis
N 1	141	%sl=21	%sf=test/bubble.sis
E	0 3	1 1	4	 %na=old-j	%mk=V	%sl=21
L		1 2	4 "1"	%mk=V
N 2	141	%sl=22	%sf=test/bubble.sis
E	1 1	2 1	4	 %na=j	%mk=V	%sl=22
L		2 2	4 "1"	%mk=V
N 3	105	%sl=22	%sf=test/bubble.sis
E	0 4	3 1	9	 %na=old-a1	%mk=V	%sl=22
E	1 1	3 2	4	 %na=j	%mk=V	%sl=22
N 4	105	%sl=22	%sf=test/bubble.sis
E	0 4	4 1	9	 %na=old-a1	%mk=V	%sl=22
E	2 1	4 2	4	%mk=V
N 5	132	%sl=22	%sf=test/bubble.sis
E	3 1	5 1	4	%mk=V
E	4 1	5 2	4	%mk=V
N 6	139	%sl=22	%sf=test/bubble.sis
E	5 1	6 1	1	%mk=V
N 7	129
E	6 1	7 1	1	%mk=V
{ Compound   8   1
G	0
E	0 1	0 1	4	%mk=V
G	0	%sl=24	%sf=test/bubble.sis
E	0 2	0 1	9	 %na=old-a1	%mk=V	%sl=25
G	0	%sl=23	%sf=test/bubble.sis
N 1	141	%sl=23	%sf=test/bubble.sis
E	0 3	1 1	4	 %na=j	%mk=V	%sl=23
L		1 2	4 "1"	%mk=V
N 2	105	%sl=23	%sf=test/bubble.sis
E	0 2	2 1	9	 %na=old-a1	%mk=V	%sl=23
E	0 3	2 2	4	 %na=j	%mk=V	%sl=23
N 3	105	%sl=23	%sf=test/bubble.sis
E	0 2	3 1	9	 %na=old-a1	%mk=V	%sl=23
E	1 1	3 2	4	%mk=V
N 4	113	%sl=23	%sf=test/bubble.sis
E	0 2	4 1	9	 %na=old-a1	%mk=V	%sl=23
E	0 3	4 2	4	 %na=j	%mk=V	%sl=23
E	3 1	4 3	4	%mk=V
E	2 1	4 4	4	%mk=V
E	4 1	0 1	9	%mk=V
} 8 1 3 0 1 2	%sl=23	%sf=test/bubble.sis
E	7 1	8 1	4	%mk=V
E	0 4	8 2	9	 %na=old-a1	%mk=V	%sl=23
E	1 1	8 3	4	 %na=j	%mk=V	%sl=23
E	1 1	0 3	4	 %na=j	%mk=V	%sl=23
E	8 1	0 4	9	 %na=a1	%mk=V	%sl=25
G	0	%sl=28	%sf=test/bubble.sis
N 1	127	%sl=29	%sf=test/bubble.sis
E	0 4	1 1	14	 %na=a1	%mk=V	%sl=28
E	1 1	0 1	9	%mk=V
} 2 4 4 0 1 2 3	%sl=17	%sf=test/bubble.sis
E	0 3	2 1	9	 %na=old-a	%mk=V	%sl=19
E	1 1	2 2	4	 %na=limit	%mk=V	%sl=20
E	2 1	0 3	9	 %na=a	%mk=V	%sl=19
E	1 1	0 4	4	 %na=limit	%mk=V	%sl=20
G	0	%sl=32	%sf=test/bubble.sis
N 1	127	%sl=33	%sf=test/bubble.sis
E	0 3	1 1	14	 %na=a	%mk=V	%sl=32
E	1 1	0 1	9	%mk=V
} 1 4 4 0 1 2 3	%sl=9	%sf=test/bubble.sis
E	0 2	1 1	9	 %na=ain	%mk=V	%sl=11
E	0 1	1 2	4	 %na=n	%mk=V	%sl=11
E	1 1	0 1	9	%mk=V
X	13	"Main"	%sl=37	%sf=test/bubble.sis
N 1	116	%sl=38	%sf=test/bubble.sis
E	0 1	1 1	9	 %na=a	%mk=V	%sl=38
N 2	120	%sl=38	%sf=test/bubble.sis
L		2 1	12 "Bubble"	%mk=V
E	1 1	2 2	4	%mk=V
E	0 1	2 3	9	 %na=a	%mk=V	%sl=38
E	2 1	0 1	9	%mk=V
