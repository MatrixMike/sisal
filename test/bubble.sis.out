DEFINE MAIN
TYPE ONEDIM = ARRAY [INTEGER]
FUNCTION BUBBLE (N:INTEGER;AIN:ONEDIM RETURNS ONEDIM)
  FOR INITIAL      A,LIMIT := AIN,N
    WHILE LIMIT > 1
      REPEAT
      LIMIT := OLD LIMIT - 1;
      A := FOR INITIAL              J := 0;
        A1 := OLD A
            WHILE J < LIMIT
              REPEAT
          J := OLD J + 1;
          A1 := IF OLD A1[J] > OLD A1[J + 1] THEN
OLD A1 [ J:OLD A1[J + 1],OLD A1[J] ]
          ELSE OLD A1
          END IF
      RETURNS VALUE OF A1
            END FOR
  RETURNS VALUE OF A
    END FOR
END FUNCTION
FUNCTION MAIN (A:ONEDIM RETURNS ONEDIM)
  BUBBLE(ARRAY_SIZE(A),A)
END FUNCTION
LIMIT
 of type:7
1
 of type:5
Output Dot in /tmp/graph-10fb9f.dot
Unexpected: "" in file: bubble.sis (line 40: char 0..0)
